src/odemis/gui/comp/overlay/world.py

```diff
@@ -832,6 +833,566 @@ def deactivate(self):
         WorldOverlay.deactivate(self)
 
 
+NONE_RULER_MODE = 0
+MOVE_RULER_MODE = 1
+EDIT_START_RULER_MODE = 2
+EDIT_END_RULER_MODE = 3
+
+class Ruler(object):
+
+    hover_margin = 10  # pixels
+
+    def __init__(self, cnvs, drag_v_start_pos, drag_v_end_pos):
```

Comment:
I think you shouldn't need drag_v_start_pos and drag_v_end_pos. 
However, what you *could* have is p_start_pos and p_stop_pos (defaulting to None). If they are defined, the v_start_pos/v_stop_pos would be immediately computed too. With them, you could pass what you currently set via `v_start_pos` and, more interestingly, you could create some Rulers from a test case.
So this would allow you to do a basic test case where you "manually" add a few Rulers to the overlay, and check that the .draw() doesn't fail.